app/src/main/java/com/iic/shopingo/services/CurrentLocationProvider.java

```diff
@@ -0,0 +1,69 @@
+package com.iic.shopingo.services;
+
+import android.content.Context;
+import android.location.Location;
+import android.os.Bundle;
+import com.google.android.gms.common.ConnectionResult;
+import com.google.android.gms.common.api.GoogleApiClient;
+import com.google.android.gms.location.LocationListener;
+import com.google.android.gms.location.LocationRequest;
+import com.google.android.gms.location.LocationServices;
+
+/**
+ * Created by assafgelber on 3/5/15.
+ */
+public class CurrentLocationProvider implements GoogleApiClient.ConnectionCallbacks, GoogleApiClient.OnConnectionFailedListener {
+  private GoogleApiClient googleApiClient;
+  private LocationListener internalListener;
+  private LocationUpdatesListener updatesListener;
+  private long requestInterval;
+
+  public CurrentLocationProvider(Context context, long requestInterval, LocationUpdatesListener listener) {
+    this.requestInterval = requestInterval;
+    this.updatesListener = listener;
+
+    googleApiClient = new GoogleApiClient.Builder(context)
+        .addApi(LocationServices.API)
+        .addConnectionCallbacks(this)
+        .build();
+    googleApiClient.connect();
+  }
+
+  @Override
+  public void onConnected(Bundle bundle) {
+    LocationRequest locationRequest = new LocationRequest();
+    locationRequest.setInterval(requestInterval);
+    locationRequest.setFastestInterval(requestInterval);
+    locationRequest.setPriority(LocationRequest.PRIORITY_BALANCED_POWER_ACCURACY);
+
+    internalListener = new LocationListener() {
```

Comment:
You can make CurrentLocationProvider implement the LocationListener interface and then you won't need the anonymous class.